generate_report.gs

```diff
@@ -0,0 +1,87 @@
+/**
+* Generates the report of most executed script in google sheets
+* @param {Object} mostExecutedScript is the array of projectIds  
+* @param {String} title is the heading for the report sheet
+* @param {Number} limitResponse is the maximum number of responses in the sheet
+* @return {String} URL of generated report
+*/
+function generateReportForMostExecutedScript(mostExecutedScript, title, limitResponse) {  
+  var timeStamp, dateTimeAsString, rowContent;
+  
+  //Create a new date
+  timeStamp = new Date();
+  dateTimeAsString = timeStamp.toString().slice(0,15);
+  var newSheet = SpreadsheetApp.create(title + " at "+ dateTimeAsString);
+  var active  = newSheet.getActiveSheet();
+  active.setName("Most Executed Script");
+  
+  // add the most executed script report
+  rowContent = ["ProjectId", "Execution Count", "Cloud Project Id"];
+  active.appendRow(rowContent);
+  var lastRow = active.getLastRow();
+  active.getRange(lastRow, 1, 1, 3).setFontSize(12).setFontWeight("BOLD");
+  for(var i = 0; i<mostExecutedScript.length && i<limitResponse; i++) {
+    rowContent = [mostExecutedScript[i].key, mostExecutedScript[i].value, mostExecutedScript[i].GCPId];
+    active.appendRow(rowContent);
+  }
+  return newSheet.getUrl();
+}
+
+
+/**
+* Generates the report of most active users in google sheets  
+* @param {Object} mostActiveUser is the array of userIds
+* @param {String} title is the heading for the report sheet
+* @param {Number} limitResponse is the maximum number of responses in the sheet
+* @return {String} URL of generated report
+*/
+function generateReportForMostActiveUsers(mostActiveUser, title, limitResponse) {  
+  var timeStamp, dateTimeAsString, rowContent;
```

Comment:
same as above and can you reuse the code instead of copy pasting same code everywhere

Reply:
Reports have different format.
That's why i kept different functions for each report type